litestar/response/base.py

```diff
@@ -385,7 +385,9 @@ def render(self, content: Any, media_type: str, enc_hook: Serializer = default_s
             if media_type == MediaType.MESSAGEPACK:
                 return encode_msgpack(content, enc_hook)
 
-            if media_type.startswith("application/json"):
+            if media_type.startswith("application/json") or (
+                media_type.startswith("application/") and media_type.endswith("+json")
+            ):
```

Comment:
This could be simplified to a regex

Reply:
Could do `re.match` with `^application/(?:.+\+)?json`

Technically, I suppose I could take into account that the string should either end after json, or have a semicolon, but that would break backwards compatibility with the old `startswith("application/json")`.

Reply:
> I do see an issue, now, however, with additional information about encoding/charset in the media type, like "application/json; charset=utf-8". I assume the "startswith" is there because of such information? That would have to be handled as well.

That would be easier to handle with a regex IMO because you don't have to rely on the `+json`/`json` part being the end of the string.

Reply:
> Could do `re.match` with `^application/(?:.+\+)?json`

Sounds good.

> Technically, I suppose I could take into account that the string should either end after json, or have a semicolon, but that would break backwards compatibility with the old `startswith("application/json")`.

Yeah, I guess we should do stricter matching here, but I don't think we can before 3.0 because it would be breaking.

Reply:
Done.